internal/controllers/core/tiltfile/api.go

```diff
@@ -47,15 +50,37 @@ type typedObjectSet map[string]object
 //
 // In the future, anything that creates objects based on the Tiltfile (e.g., FileWatch specs,
 // LocalServer specs) should go here.
-func UpdateOwnedObjects(ctx context.Context, client ctrlclient.Client, tlr tiltfile.TiltfileLoadResult, mode store.EngineMode) error {
-	apiObjects := toAPIObjects(tlr, mode)
+func updateOwnedObjects(ctx context.Context, client ctrlclient.Client, nn types.NamespacedName,
+	tf *v1alpha1.Tiltfile, tlr *tiltfile.TiltfileLoadResult, mode store.EngineMode) error {
+	apiObjects := toAPIObjects(tf, tlr, mode)
+
+	// Propagate labels and owner references from the parent tiltfile.
+	for _, objMap := range apiObjects {
+		for _, obj := range objMap {
+			err := controllerutil.SetControllerReference(tf, obj, client.Scheme())
+			if err != nil {
+				return err
+			}
+
+			if len(tf.Spec.Labels) > 0 {
+				labels := obj.GetLabels()
+				if labels == nil {
+					labels = make(map[string]string)
+				}
+				for k, v := range tf.Spec.Labels {
+					labels[k] = v
```

Comment:
I don't have a compelling argument on either side - just checking that this is desired behavior to override any overlapping labels?

Reply:
good question! ya, I changed it so that labels from tiltfile execution take precedence. I also don't really have a compelling argument either way, but this seems like more what i'd expect